exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/MetricAdapter.java

```diff
@@ -275,14 +276,14 @@ private static Sample createSample(
   }
 
   private static io.prometheus.client.exemplars.Exemplar toPrometheusExemplar(Exemplar exemplar) {
-    // TODO: Keep filtered attributes for prometheus?
     if (exemplar.getSpanId() != null && exemplar.getTraceId() != null) {
       return new io.prometheus.client.exemplars.Exemplar(
           valueOf(exemplar),
-          exemplar.getRecordTimeNanos(),
-          "traceid",
+          // Convert to ms for prometheus, truncate nanosecond preceision.
```

Comment:
May I ask how you're getting ns resolution timestamps? System.currentTimeMillis() is all that Java provides in this area.

Reply:
`Instant.now` gets us microseconds on Java 9+, we use it here

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/main/java9/io/opentelemetry/sdk/internal/Java9VersionSpecific.java#L39

We indeed don't have nanos for arbitrary wall time but do have the notion of "nanos anchored to the beginning of a Span"

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/main/java/io/opentelemetry/sdk/internal/MonotonicClock.java

So OTel uses nanos for all its timestamps as convention. For exemplars with a span ID, it seems the timestamp could be derived from the span start time, but that could make it out of sync with the timestamp of the raw data recording so I guess it's not something we'd do.

Reply:
We haven't actually figured out how to record the timestamp just yet.   OTLP / our metrics data model consistently wants it in `ns`, so that's how we plan to track it here.   As @anuraaga states, there are options in Otel for finer-grained resolution, particularly for exemplars as we sample those and COULD use the `MonotonicClock` of the associated `Span`.

This PR does NOT introduce sampling of exemplars, only the export from the data model.